pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -1208,11 +1238,11 @@ func (n *NetworkPolicyController) syncInternalNetworkPolicy(key string) error {
 	// Lock the internal NetworkPolicy store as we may have a case where in the
 	// same internal NetworkPolicy is being updated in the NetworkPolicy UPDATE
 	// handler.
-	n.internalNetworkPolicyMutex.Lock()
+	n.internalNetworkPolicyMutex.RLock()
```

Comment:
this seems to be an orthogonal change, do you think you can put it in a separate PR that we can merge right away?

Reply:
actually we do update the store at L1271.. should this be R?